src/main/java/io/redisearch/client/Client.java

```diff
@@ -383,13 +386,24 @@ public boolean addDocument(Document doc, AddOptions options) {
 	    	return results;
     	}
     }
-    
+
+    private void addDocument(Document doc, AddOptions options, Pipeline p) {
+        ArrayList<byte[]> args = addDocumentArgs(doc, options);
+        p.sendCommand(commands.getAddCommand(), args.toArray(new byte[args.size()][]));
+
+    }
+
     private BinaryClient addDocument(Document doc, AddOptions options, Jedis conn) {
+        ArrayList<byte[]> args = addDocumentArgs(doc, options);
+        return sendCommand(conn, commands.getAddCommand(), args.toArray(new byte[args.size()][])); 
+    }
+
+    private ArrayList<byte[]> addDocumentArgs(Document doc, AddOptions options) {
```

Comment:
@filipecosta90 if you can return byte[args.size()][] and not an ArrayList you might be able to save the ArrayList allocation

Reply:
@gkorland done